extensions/adobe/experience/campaign/experienceevent.schema.json

```diff
@@ -0,0 +1,259 @@
+          "default": false
+        },
+        "https://ns.adobe.com/experience/campaign/delivery/messagesClass": {
+          "title": "Message Class",
+          "type": "integer",
+          "meta:enum": {
+            "oneTime": "Marketing (One Time)",
+            "continuous": "Recurring",
+            "messageCenter": "Transactional (Event-based)"
+          },
+          "description":
+            "Marketing classification of the messages sent through that delivery.\n",
+          "default": 0
+        },
+        "https://ns.adobe.com/experience/campaign/delivery/template": {
+          "title": "Template",
+          "type": "integer",
+          "description":
+            "The delivery template used to initialize this delivery.\n\nThe type the template used in Adobe Campaign can be identified using the `messageClass`.\n\n* `messageClass` = `oneTime`: the template is standard delivery template.\n* `messageClass` = `continuous`: the template is a recurring delivery.\n* `messageClass` = `transactional`: the template is a transactionnal message template.\n"
+        },
+        "https://ns.adobe.com/experience/campaign/link/id": {
+          "title": "Tracking URL ID",
+          "type": "string",
+          "description": "Unique Identifier of the Link"
+        },
+        "https://ns.adobe.com/experience/campaign/link/description": {
+          "title": "Tracking URL Description",
+          "type": "string",
+          "description": "A human-friendly description of the Link."
+        },
+        "https://ns.adobe.com/experience/campaign/link/url": {
+          "title": "URL",
+          "format": "uri",
+          "type": "string",
+          "description": "URL of the Link."
+        },
+        "https://ns.adobe.com/experience/campaign/link/occurrence": {
+          "title": "Occurrence",
+          "type": "integer",
+          "description":
+            "URL occurrence index in the content (first is 1). When the same URL is available several times in the same content, its is necessary to track which occurrence of the URL has been clicked on.",
+          "default": 1,
+          "minimum": 1,
+          "maximum": 32767
+        },
+        "https://ns.adobe.com/experience/campaign/link/tags": {
+          "title": "Tags",
+          "type": "array",
+          "description":
+            "A list of tags qualifying the link. The URL category is one of those tags.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "https://ns.adobe.com/experience/campaign/message/offerProposed": {
+          "title": "Offers",
+          "type": "array",
+          "description": "Offers displayed in the message.",
+          "items": {
+            "$ref":
+              "https://ns.adobe.com/xdm-extensions/adobe/experience/campaign/offer"
+          }
+        },
+        "https://ns.adobe.com/experience/campaign/message/offerClicked": {
+          "title": "Offer Clicked",
```

Comment:
aren't you missing the type here?

Reply:
This is a reference to the schema `https://ns.adobe.com/xdm-extensions/adobe/experience/campaign/offer`.
I might be missing something, but I don't think type is required here, can you help?

Reply:
Yes, the `$ref` is giving you a type.

Reply:
👍